kairos-cli/src/commands/deposit.rs

```diff
@@ -15,14 +20,31 @@ pub struct Args {
     private_key_path: PrivateKeyPathArg,
 }
 
-pub fn run(args: Args) -> Result<String, CliError> {
-    let _amount: u64 = args.amount.field;
-    let _signer =
+pub fn run(args: Args, kairos_server_address: Url) -> Result<String, CliError> {
+    let amount: u64 = args.amount.field;
+    let signer =
         Signer::from_private_key_file(args.private_key_path.field).map_err(CryptoError::from)?;
 
-    // TODO: Create transaction and sign it with `signer`.
+    let client = reqwest::Client::new();
+    let public_key = signer.to_public_key()?;
 
-    // TODO: Send transaction to the network, using Rust SDK.
+    let payload = SigningPayload::new_deposit(amount)
+        .try_into()
+        .expect("Failed serialize the deposit payload to bytes");
+    let signature = signer.sign(&payload)?;
+    let deposit_request = PayloadBody {
+        public_key,
+        payload,
+        signature,
+    };
```

Comment:
I think it would be better to move `PayloadBody` construction into `submit_transaction_request()` at `kairos-cli/src/client.rs`. CLI subcommand - deposit in this case - should not be aware of REST details, client should.

Reply:
The api of `submit_transaction_request` should in future depending on the type of transaction we are doing ideally change the endpoint to either `deposit`, `transfer` and `withdraw` and could be used for all of these types.
If I make the PayloadBody construction part of the `submit_transaction_request` the api would be
```Rust
submit_transaction_request(public_key: Vec<u8>, payload: Vec<u8>, signature: Vec<u8>)
```
which could accidentally be invoked like:
```Rust
submit_transaction_request(payload, signature, public_key)
```
Which is not as safe as making the `PayloadBody` construction part of the function I would argue.

Reply:
Not really. In the future we should have single endpoint for handling all types of transactions - https://github.com/cspr-rad/kairos/issues/97. Also, once full transaction format is adopted in server, then it would become `submit_transaction_request(full_tx)`.

However, when #23 is done there will be no REST client in `kairos-cli` at all. So I consider this is only temporary code, no reason to discuss it longer.